build.sbt

```diff
@@ -27,6 +27,11 @@ scalaVersion := scala212
 crossScalaVersions := supportedScalaVersions
 
 libraryDependencies ++= Seq(
+  if (scalaVersion.value == scala211 || sparkVersion.value == "2.4.2") {
```

Comment:
I wonder if instead of `sparkVersion.value == "2.4.2"` we should have some higher-level variable like `val sparkMajorVersion = 2` or something like this.  Then we don't have to worry about things breaking if we change `sparVersion` to `2.4.4` or some other value later on.

Reply:
Thanks! I'll refer this comment later for Spark 3.0 :)